ons_alpha/bulletins/models.py

```diff
@@ -121,6 +124,14 @@ def get_context(self, request, *args, **kwargs):
         context["toc"] = self.toc
         return context
 
+    def save(self, *args, **kwargs):
+        # Set the previous_version field for corrections
+        if self.revisions.exists() and not self.previous_version:
+            latest_revision = self.revisions.order_by("-created_at").first()
+            if latest_revision:
+                self.previous_version = latest_revision.page
```

Comment:
Issue: I believe `latest_revision.page == self` here. The relation will need to be to the revision itself.